Assets/Scripts/SS3D/Systems/Traits/Trait.cs

```diff
@@ -9,7 +9,7 @@ namespace SS3D.Systems
     public class Trait : ScriptableObject
     {
         // Hash used for identification
-        protected int hash;
+        public int hash;
```

Comment:
that was quickly changed for syncing but that's a mistake, instead, use a serializer for traits, getting those field with public getter properties.